src/GslCore/Snapgene.fs

```diff
@@ -0,0 +1,127 @@
+﻿module snapgene
+
+open System.IO
+open System
+open commonTypes
+open constants
+open Amyris.Bio.utils
+open utils
+open Amyris.Dna
+open Genbank
+        
+/// Emit Snapgene (genbank) format
+///  outDir : string   tag: string  prefix for files  assemblies : List of AssemblyOut
+let dumpSnapgene (outDir:string) (tag:string) (assemblies : DnaAssembly list) (primers:DivergedPrimerPair list list option) =
+    // pair up the primer list (if they exist) with the matching assembly
+    let assemWithPrimers = ( 
+                                match primers with
+                                | Some p -> p
+                                | None -> List.init assemblies.Length (fun _ -> [])
+                            ) 
+                            |> List.zip assemblies
+
+    for (a,primers) in assemWithPrimers do
+
+        let path = sprintf "%s.%d.dna" tag (match a.id with None -> failwith "ERROR: unassigned assembly id" | Some(i) -> i )
+                        |> opj outDir
+        printf "Writing snapgene output to dir=%s tag=%s path=%s\n" outDir tag path
+
+
+        use outF = new StreamWriter(path)
+        let w (s:string) = outF.Write(s)
+        
+        let locusName = sprintf "%s_snapgene_output" tag
+        let totLength = a.dnaParts |> List.map (fun p -> p.dna.Length) |> Seq.sum
+        let now = DateTime.Now
+        sprintf "LOCUS       %-22s %d bp ds-DNA     linear   SYN %2d-%s-%d
+DEFINITION  .
+ACCESSION   .
+VERSION     .
+KEYWORDS    .
+SOURCE      synthetic DNA construct
+  ORGANISM  synthetic DNA construct
+COMMENT
+REFERENCE   1  (bases 1 to %d)
+  AUTHORS   .
+  TITLE     Direct Submission
+  JOURNAL   Exported %s %2d, %d from SnapGene Viewer 4.0.5
```

Comment:
I was really liking snapgene till I generated this output and discovered after hours of painstaking experimentation that unless you get the header perfectly like this (you can't even use exported instead of Exported in the first line,   it refuses to observe the coloring directives.  I think they must do a big match on this to decide it's a snapgene ready genbank+ format.  I'll give them feedback but for now this is how it has to be ;(

Reply:
Oh my...

Reply:
They emailed me back (promptly) with a more precise specification of header requirements, so I've relaxed the output a bit and made it more useful.  They are grepping on stupid things (and will further relax it in next release)